scubagoggles/rego/Drive.rego

```diff
@@ -0,0 +1,1588 @@
+tests contains {
+    "PolicyId": DriveId2_2,
+    "Criticality": "Should",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs2_2, NonCompliantGroups2_2),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs2_2, "NonCompliantGroups": NonCompliantGroups2_2},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check2_2_OK
+    Conditions := {count(NonCompliantOUs2_2) == 0, count(NonCompliantGroups2_2) == 0 }
+    Status := (false in Conditions) == false
+}
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.2.3
+#--
+
+DriveId2_3 := utils.PolicyIdWithSuffix("GWS.DRIVEDOCS.2.3")
+
+LogMessage2_3 := "Shared Drive Creation new_team_drive_restricts_direct_access"
+
+Check2_3_OK if {
+    not PolicyApiInUse
+    events := utils.FilterEventsOU(LogEvents, LogMessage2_3, utils.TopLevelOU)
+    count(events) > 0
+}
+
+Check2_3_OK if {PolicyApiInUse}
+
+NonComplianceMessage2_3 := "Users who aren't shared drive members are not allowed to be added to files."
+
+NonCompliantOUs2_3 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage2_3
+}
+if {
+    not PolicyApiInUse
+    some OU in utils.OUsWithEvents
+    Events := utils.FilterEventsOU(LogEvents, LogMessage2_3, OU)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    contains("true", LastEvent.NewValue) == false
+    LastEvent.NewValue != "DELETE_APPLICATION_SETTING"
+}
+NonCompliantGroups2_3 contains {
+    "Name": Group,
+    "Value": NonComplianceMessage2_3
+}
+if {
+    not PolicyApiInUse
+    some Group in utils.GroupsWithEvents
+    Events := utils.FilterEventsGroup(LogEvents, LogMessage2_3, Group)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    contains("true", LastEvent.NewValue) == false
+    LastEvent.NewValue != "DELETE_APPLICATION_SETTING"
+}
+
+NonCompliantOUs2_3 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage2_3
+}
+if {
+    some OU, settings in input.policies
+    DriveEnabled(OU)
+    nonMemberAccess := settings.drive_and_docs_shared_drive_creation.allowNonMemberAccess
+    nonMemberAccess != true
+}
+
+tests contains {
+    "PolicyId": DriveId2_3,
+    "Criticality": "Shall",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    not PolicyApiInUse
+    DefaultSafe := true
+    not Check2_3_OK
+}
+
+tests contains {
+    "PolicyId": DriveId2_3,
+    "Criticality": "Shall",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs2_3, NonCompliantGroups2_3),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs2_3,
+        "NonCompliantGroups": NonCompliantGroups2_3},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check2_3_OK
+    Conditions := {count(NonCompliantOUs2_3) == 0, count(NonCompliantGroups2_3) == 0 }
+    Status := (false in Conditions) == false
+}
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.2.4
+#--
```

Comment:
The report fails you no matter what you select for "Allow viewers and commenters to download, print, and copy files"

Reply:
![image](https://github.com/user-attachments/assets/bcb6c8b0-1a74-4a65-ae53-28250996c9c0)

Reply:
The org we use for testing uses Google Workspace Enterprise Plus, so that isn't the issue.

Reply:
After implementing Google's policy "reduction" process, this is working for me
when I change settings in the Admin UI.